src/components/Notes/Note.jsx

```diff
@@ -45,8 +45,8 @@ const Note = ({
       setCurrNote(note.body);
       return;
     }
-    const newNotes = notes.map((n, idx) => {
-      if (idx === currNoteIdx) {
+    const newNotes = notes.map((n) => {
+      if (n._id === note._id) {
```

Comment:
theres a lot of array reversing happening in the front-end, so i simplified this to just check for IDs instead